notebooks/MAST/HSC/CASJOBS/HCV_CASJOBS/HCV_casjobs_demo.ipynb

```diff
@@ -0,0 +1,995 @@
+    "# we are using some features not in version 0.0.1\n",
+    "from pkg_resources import get_distribution\n",
+    "from packaging.version import Version as V\n",
+    "\n",
+    "assert V(get_distribution(\"mastcasjobs\").version) >= V('0.0.2'), \"\"\"\n",
+    "A newer version of mastcasjobs is required.\n",
+    "Update mastcasjobs to current version using this command:\n",
+    "pip install --upgrade git+git://github.com/rlwastro/mastcasjobs@master\n",
+    "\"\"\"\n",
+    "\n",
+    "import mastcasjobs\n",
+    "\n",
+    "# set width for pprint\n",
+    "astropy.conf.max_width = 150"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# set universal matplotlib parameters\n",
+    "plt.rcParams.update({'font.size': 16})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "HSCContext = \"HSCv3\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set up Casjobs environment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import getpass\n",
+    "if not os.environ.get('CASJOBS_USERID'):\n",
+    "    os.environ['CASJOBS_USERID'] = input('Enter Casjobs UserID:')\n",
+    "if not os.environ.get('CASJOBS_PW'):\n",
+    "    os.environ['CASJOBS_PW'] = getpass.getpass('Enter Casjobs password:')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### MAST Functions\n",
+    "\n",
+    "The `resolve(name)` function uses the MAST Name Resolver (which relies on both SIMBAD and NED) to get the RA, Dec position for an object."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def resolve(name):\n",
+    "    \"\"\"Get the RA and Dec for an object using the MAST name resolver\n",
+    "    \n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    name (str): Name of object\n",
+    "\n",
+    "    Returns RA, Dec tuple with position\n",
+    "    \"\"\"\n",
+    "\n",
+    "    resolverRequest = {'service': 'Mast.Name.Lookup',\n",
+    "                       'params': {'input': name,\n",
+    "                                  'format': 'json'}}\n",
+    "    resolvedObjectString = mastQuery(resolverRequest)\n",
+    "    resolvedObject = json.loads(resolvedObjectString)\n",
+    "    # The resolver returns a variety of information about the resolved object, \n",
+    "    # however for our purposes all we need are the RA and Dec\n",
+    "    try:\n",
+    "        objRa = resolvedObject['resolvedCoordinate'][0]['ra']\n",
+    "        objDec = resolvedObject['resolvedCoordinate'][0]['decl']\n",
+    "    except IndexError:\n",
+    "        raise ValueError(f\"Unknown object '{name}'\")\n",
+    "    return (objRa, objDec)\n",
```

Comment:
This smells like re-inventing the wheel, and is something we should discourage our users from doing.  Instead of using a custom mast-resolver, I suggest the following:
```
from astropy.coordinates import SkyCoord

coord = SkyCoord.from_name(name)
```
and then use the `coord` object in place of carrying ra and dec around. Failing that, one should instead use the `astroquery` package to do this kind of name resolution.

Reply:
Replaced